os/arch/arm/src/common/up_mpuinit.c

```diff
@@ -50,11 +50,56 @@
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
+struct mpu_regs g_mpuregs;
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: mpu_regs_initialize
+ *
+ * Description:
+ *   Initialize the MPU region numbers.
+ *
+ ****************************************************************************/
+
+void mpu_regs_initialize(FAR struct mpu_regs *mpu)
+{
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
```

Comment:
We should use another way to get the value for each.
Currently 2nd block concerns the config of 1st block. This makes code complex.
In 2nd block, only the config of 2nd block should be concerned.
How about below?
```c
uint8_t offset;

offset = mpu->board...;

#ifdef config_common
#ifdef config_reload_time
offse += 3;
#else
offset += 1;
#endif
mpu->common.. = offset;
#endif

#ifdef config_app_separation
#ifdef config_reload_time
offse += 3;
#else
offset += 1;
#endif
mpu->app_regs = offset;
#endif
```

Reply:
Done